.github/workflows/release.yml

```diff
@@ -10,23 +13,71 @@ permissions:
 
 jobs:
   verify-version:
+    name: Verify that version that triggered this workflow is greater than most recent  release
     runs-on: ubuntu-latest
+    outputs:
+      versionIsValid: ${{ steps.validVersion.outputs.versionIsValid }}
+      version: ${{ steps.validVersion.outputs.version }}
     steps:
       - uses: actions/checkout@v4
 
+      - uses: actions/setup-node@v3
+        with:
+          cache: 'npm'
+          cache-dependency-path: scripts/package-lock.json
+      - run: npm ci
+        working-directory: scripts
+
       - name: Get version from Cargo.toml
         id: lookupVersion
         uses: mikefarah/yq@a198f72367ce9da70b564a2cc25399de8e27bf37
         with:
           cmd: yq -oy '"v" + .workspace.package.version' 'Cargo.toml'
 
-      - name: Version check
-        if: steps.lookupVersion.outputs.result != github.ref_name
-        run: echo "Project version in Cargo.toml does not match the tag" && exit 1
+      - name: Get version from the latest releases
+        id: lookupVersionRelease
+        uses: pozetroninc/github-action-get-latest-release@master
+        with:
+          owner: foundry-rs
+          repo: starknet-foundry
+          excludes: prerelease, draft
+
+      - name: Compare versions
+        id: validVersion
+        run: |
+          RELEASE_VERSION=${{ steps.lookupVersionRelease.outputs.release }}
+          COMMIT_VERSION=${{ steps.lookupVersion.outputs.result }}
+          echo "Project version from newest release = $RELEASE_VERSION"
+          echo "Project version from this commit = $COMMIT_VERSION"
+          IS_VALID=$(node ./scripts/compareVersions.js $RELEASE_VERSION $COMMIT_VERSION)
```

Comment:
was there a specific reason to write js script, and not just use bash?

Reply:
Handling semver versions in bash seemed much more complicated than using `semver` library with node. If there's some simple solution for bash, we can change to that.

Reply:
not a problem, I was just curious